### PR TITLE
Fix the bugs

### PR DESCRIPTION
**Pull request by Haoyuli2002**
Fix the bugs of #190, #159 #176 #151 #129

### CODE REVIEW

File: src/main/java/essenmakanan/recipe/RecipeList.java
Comment (charkty) [inline src/main/java/essenmakanan/recipe/RecipeList.java:1]: The case of when s/ is empty is properly handled. However, is the case of when there is no string attached behind edit i/RECIPE_TITLE handled? #190

File: src/main/java/essenmakanan/command/AddRecipeCommand.java [hunk 1]
```diff
@@ -157,7 +157,10 @@ public void addValidRecipe() throws EssenFormatException {
             }
             slashIndex = nextSlashIndex;
         }
-
+        if (recipeTitle.length() == 0) {
```
Comment (charkty) [inline src/main/java/essenmakanan/command/AddRecipeCommand.java:160]: This works and solves #159! But is recipeTitle.isEmpty() more appropriate for strings?

File: src/main/java/essenmakanan/command/EditRecipeCommand.java [hunk 2]
```diff
@@ -40,4 +45,14 @@ public void executeCommand() {
 
     }
 
+    private static List<String> removeWhiteSpaces(String[] splitDetails) {
+        List<String> detailList = new ArrayList();
+        for (String detail : splitDetails) {
+            if (detail.length() > 0) {
+                detailList.add(detail);
```
Comment (charkty) [inline src/main/java/essenmakanan/command/EditRecipeCommand.java:52]: Do you need to use detail.trim() to remove the trailing and leading whitespaces in order to solve #176 ?

File: src/main/java/essenmakanan/ui/Ui.java [hunk 3]
```diff
@@ -131,9 +135,14 @@ public static void printStartRecipeMessage(IngredientList missingIngredients,
     }
 
     public static void printAllRecipes(RecipeList recipes) {
-        System.out.println("Here's a list of your recipes!");
-        recipes.listRecipeTitles();
-        drawDivider();
+        if (recipes.getRecipes().size() == 0) {
+            System.out.println("Your Recipe List is empty right now, please create your own recipe first :D!");
```
Comment (charkty) [inline src/main/java/essenmakanan/ui/Ui.java:139]: Very concise handling of the first part of #151 ! However, did you miss out on error messages for when recipe list is empty and the user types view r/1? An error message should be printed in that case too.

**Review by charkty — COMMENTED** [review]
There are a few bugs that remain unresolved.

File: src/main/java/essenmakanan/command/AddRecipeCommand.java (hunk 1 above)
Comment (Haoyuli2002) [reply src/main/java/essenmakanan/command/AddRecipeCommand.java:160]: Thanks, will modify later

File: src/main/java/essenmakanan/command/EditRecipeCommand.java (hunk 2 above)
Comment (Haoyuli2002) [reply src/main/java/essenmakanan/command/EditRecipeCommand.java:52]: No, we have already used the white space to split the string, so for example "edit /r bread     n/flavor"
will be split into the array ["bread", "", "", "", "", "", "n/flavor"], I just want to delete this empty string in this case

File: src/main/java/essenmakanan/ui/Ui.java (hunk 3 above)
Comment (Haoyuli2002) [reply src/main/java/essenmakanan/ui/Ui.java:139]: Will check later

File: src/main/java/essenmakanan/command/ViewSpecificRecipeCommand.java [hunk 4]
```diff
@@ -19,6 +19,10 @@ public ViewSpecificRecipeCommand(RecipeList recipes, String input) {
     @Override
     public void executeCommand() {
         try {
+            if (recipes.getRecipes().size() == 0) {
+                System.out.println("You haven't added any recipes yet, please add some recipes first!");
+                return;
+            }
             int recipeIndex = RecipeParser.getRecipeIndex(recipes, input);
```
Comment (charkty) [inline src/main/java/essenmakanan/command/ViewSpecificRecipeCommand.java:26]: #151 handled!

**Review by charkty — APPROVED** [review]
LGTM

**Pull request merged**